cmd/skopeo/sync.go

```diff
@@ -509,9 +513,15 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) error {
 		return err
 	}
 
+	ctx, cancel := opts.global.commandTimeoutContext()
+	defer cancel()
+
 	sourceArg := args[0]
-	srcRepoList, err := imagesToCopy(sourceArg, opts.source, sourceCtx)
-	if err != nil {
+	var srcRepoList []repoDescriptor
+	if err = retryIfNecessary(ctx, func() error {
+		srcRepoList, err = imagesToCopy(sourceArg, opts.source, sourceCtx)
```

Comment:
(This applies a single `retryOpts.maxRetry` to the whole call, even if it contacts many registries and checks many repositories. I’m completely undecided whether it’s what the users expect; for now, this works well enough and gets us to 100% retry coverage.)